web/src/OverviewTable.tsx

```diff
@@ -478,29 +536,53 @@ function resourceTypeLabel(r: UIResource): string {
   return "Unknown"
 }
 
+function hasWidgets(view: Proto.webviewView): boolean {
+  return !!view.uiButtons?.length
+}
+
+function viewToRowValues(
+  view: Proto.webviewView,
+  logStore: LogStore
+): RowValues[] {
+  return (
+    view.uiResources?.map((r) =>
+      uiResourceToCell(r, view.uiButtons || [], logStore)
+    ) || []
+  )
+}
+
 export default function OverviewTable(props: OverviewTableProps) {
   let logStore = useLogStore()
-  const data = React.useMemo(
-    () =>
-      props.view.uiResources?.map((r) => uiResourceToCell(r, logStore)) || [],
-    [props.view.uiResources]
-  )
+  const data = React.useMemo(() => viewToRowValues(props.view, logStore), [
+    props.view.uiResources,
+    props.view.uiButtons,
+  ])
 
   const {
     getTableProps,
     getTableBodyProps,
     headerGroups,
     rows,
     prepareRow,
+    columns,
   } = useTable(
     {
-      columns,
+      columns: columnDefs,
       data,
       autoResetSortBy: false,
     },
     useSortBy
   )
 
+  // if there are no widgets, hide the widgets column
+  // TODO(matt) is this actually what we want?
+  const widgetsVisible = hasWidgets(props.view)
```

Comment:
this makes sense to me, but i'll let @SurbhiSGupta and others chime in too!

Reply:
I'm happy to change this later, but don't think it matters enough either way to block merging on